test/function/samples/logical-expressions-literal-value/main.js

```diff
@@ -0,0 +1,7 @@
+var MyEnum =
+	(unknownGlobal,
+	(MyEnum2 => {
+		MyEnum2['foo'] = 'FOO';
+		MyEnum2['bar'] = 'BAR';
+		return MyEnum2;
+	})(MyEnum || {}));
```

Comment:
How about adding an assertion at the end

```ts
assert.strictEqual(MyEnum.foo, 'FOO');
```

Then we definitely have the property access, and this does not change anything about the original bug.